src/display.rs

```diff
@@ -88,12 +88,48 @@ impl From<renderer::Error> for Error {
     }
 }
 
+pub struct VisualBell {
+    /// Visual bell duration
+    duration: u16,
+
+    /// The last time the visual bell rang, if at all
+    last_rang: Option<::std::time::Instant>,
+}
+
+impl VisualBell {
+    /// Ring the visual bell, and return its value.
+    pub fn ring(&mut self) -> f64 {
+        let now = ::std::time::Instant::now();
+        self.last_rang = Some(now);
+        self.value(self.last_rang)
+    }
+
+    /// Get the value of the visual bell. This value ramps down from 1.0 to 0.0
+    /// at a rate determined by the bell's duration.
+    pub fn value(&self, when: Option<::std::time::Instant>) -> f64 {
+        match self.last_rang {
+            Some(earlier) => {
+                let now = when.unwrap_or(::std::time::Instant::now());
+                if now < earlier {
+                    return 0.0;
+                }
+                let duration = now.duration_since(earlier);
+                let elapsed_ms = duration.as_secs() as f64 * 1_000f64 +
+                                 duration.subsec_nanos() as f64 / 1_000_000f64;
+                (1.0 - elapsed_ms / self.duration as f64).max(0.0)
```

Comment:
Linear is definitely one way to go, but I think it might be good to consider some sort of "ease out" function where  most of the intensity disappears quickly. A simple example of this would be squaring or cubing the normalized elapsed time.